File: python/questions/findMinArrowShots.py
from __future__ import annotations
import math

class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        ans = 0
        pre = -math.inf
        for start, end in points:
            if start > pre:
                ans += 1
                pre = end
        return ans
        

if __name__ == "__main__":
    a = []
    a.append([1, 9])
    a.append([7, 8])
    a.append([3, 5])
    a.append([4, 6])
    b = Solution()
    print(b.findMinArrowShots(a))

File: c/questions/findMinArrowShots.c
#include <stdlib.h>
#include <limits.h>

int cmp(const void* a, const void* b) {
  int ae = (*(int**)a)[1];
  int be = (*(int**)b)[1];
  return (ae > be) - (ae < be);
}

int findMinArrowShots(int** points, int pointsSize, int* pointsColSize) {
  qsort(points, pointsSize, sizeof(int*), cmp);
  int ans = 0;
  int pre = INT_MIN;
  for (int p=0; p<pointsSize; p++) {
    if (points[p][0] > pre) {
      ans++;
      pre = points[p][1];
    }
  }
  return ans;
}
